### ISSUE TITLE
FieldsCamelCasedWithUnderscoreCodeFixProvider is a large cause of GC pressure in the wild (up 1695 mb/second)

### ISSUE THREAD

**Issue by davkean** [ni/csharp#77]
Hey folks, I'm from the Visual Studio performance team. Our performance telemetry has caught an issue in the wild where FieldsCamelCasedWithUnderscoreCodeFixProvider produces extremely large amounts of objects on the Gen0 heap, putting large amount of pressure on the GC. This ultimately shows up as responsiveness and freezing issues within Visual Studio. 

In one trace we've captured, we can see over a 60 second period, FieldsCamelCasedWithUnderscoreCodeFixProvider allocated close to 77 GB (!) of memory, causing GC pause time to push over 30% of wall clock time.

This consisted of four main allocations here: https://github.com/ni/csharp-styleguide/blob/main/src/NationalInstruments.Analyzers/Style/FieldsCamelCasedWithUnderscoreCodeFixProvider.cs#L90-L116

Type|Cause|Notes
---|---|---
System.String|`baseName + suffix.ToString(CultureInfo.InvariantCulture)`|
System.String|`suffix.ToString(CultureInfo.InvariantCulture)`|
System.Func`2[Microsoft.CodeAnalysis.ISymbol,System.Boolean]|`x => string.Equals(x.Name, name, StringComparison.Ordinal)`|
ImmutableArray<ISymbol>.EnumeratorObject[Microsoft.CodeAnalysis.ISymbol]|`Any(x => string.Equals(x.Name, name, StringComparison.Ordinal))`|Boxing the ImmutableArray<T>'s struct enumerator

I'm not in the right position to advise on the best way to resolve this but I've reached out to the Roslyn team to provide input and will get them to reply to this.

**Comment by mavasani** [comment]
Seems like the loop condition below should not have the `!`. We want to loop only while there is a member symbol with matching `name`.

```csharp
            var suffix = 2;
            var name = baseName;

            while (!symbols.Any(x => string.Equals(x.Name, name, StringComparison.Ordinal)))
            {
                name = baseName + suffix.ToString(CultureInfo.InvariantCulture);
                suffix++;
            }

            return name;
```

**Comment by jryckman** [comment]
Thanks for raising awareness of this issue. I'll see if we can patch this in the analyzer soon.

**Comment by pakdev** [comment]
Fixed with #78
